pkg/client/userd/grpc.go

```diff
@@ -61,7 +61,8 @@ func (s *service) withSession(c context.Context, callName string, f func(context
 			return
 		}
 		defer func() { err = callRecovery(recover(), err) }()
-		err = f(s.sessionContext, s.session)
+		ctx := dgroup.WithGoroutineName(s.sessionContext, "-"+callName)
```

Comment:
I think you must do what `callCtx` does here and use an integer suffix, to be able to differentiate between parallel calls.